packages/ra-auth-cognito/src/authProvider.ts

```diff
@@ -185,7 +185,9 @@ export const CognitoAuthProvider = (
                 }
                 user.getSession((err, session) => {
                     if (err) {
-                        return reject(err);
+                        return redirectToOAuthIfNeeded(
+                            new HttpError('No user', 401)
+                        );
```

Comment:
Fix seems good.

But now I'm realizing that the `redirectToOAuthIfNeeded` fn actually does not use the provided `error` param. Maybe we should clean that up.